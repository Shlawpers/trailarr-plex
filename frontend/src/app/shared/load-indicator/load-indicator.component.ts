import {ChangeDetectionStrategy, Component, HostBinding} from '@angular/core';

@Component({
  selector: 'app-load-indicator',
  template: `<div class="loading-bar"></div>
    <div class="loading-bar"></div>
    <div class="loading-bar"></div>
    <div class="loading-bar"></div>`,
  changeDetection: ChangeDetectionStrategy.OnPush,
  imports: [],
})
export class LoadIndicatorComponent {
  @HostBinding('class') hostClass = 'loading-wave';
}
