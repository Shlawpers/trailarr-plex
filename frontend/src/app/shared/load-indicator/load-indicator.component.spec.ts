import {MockBuilder, MockedComponentFixture, MockRender} from 'ng-mocks';
import {LoadIndicatorComponent} from './load-indicator.component';

describe('LoadIndicatorComponent', () => {
  let fixture: MockedComponentFixture<LoadIndicatorComponent>;

  beforeEach(() => MockBuilder(LoadIndicatorComponent));

  beforeEach(() => {
    fixture = MockRender(LoadIndicatorComponent);
    fixture.detectChanges();
  });

  it('has instance', () => expect(fixture.componentInstance).toBeTruthy());

  it('renders', () => expect(fixture).toMatchSnapshot());
});
