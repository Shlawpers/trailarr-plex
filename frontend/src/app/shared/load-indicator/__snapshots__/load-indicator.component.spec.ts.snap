// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadIndicatorComponent renders 1`] = `
<mock-render>
  <app-load-indicator
    class="loading-wave"
  >
    <div
      class="loading-bar"
    />
    <div
      class="loading-bar"
    />
    <div
      class="loading-bar"
    />
    <div
      class="loading-bar"
    />
  </app-load-indicator>
</mock-render>
`;
